scripts/tpu-cleanup.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -x
+set -u
+
+zones=("us-central1-b" "us-central2-b" "europe-west4-a")
+
+for zone in ${zones[@]}; do
+  region=${zone%-*}
+  cluster=xl-ml-test-$region
+  gcloud container clusters get-credentials $cluster --region $region --project xl-ml-test
+
+  # Don't delete any TPUs for pods that are currently running
+  pods_to_save=$(mktemp)
+  for namespace in default automated; do
+    for status in Running Pending; do
+      kubectl get pods -n $namespace --field-selector status.phase=$status -o jsonpath='{.items[*].metadata.uid}' \
+        | tr ' ' '\n' \
+          >> $pods_to_save
+      # Ensure there is a newline between outputs
+      echo "" >> $pods_to_save
```

Comment:
At the cost of readability,  `sed -ie '$a\' $pods_to_save` will append a newline if absent. Then line 25 could be removed IIUC

https://stackoverflow.com/questions/23055831/add-new-line-character-at-the-end-of-file-in-bash